### DISCUSSION TITLE
Luet probleme upgrade system

### DISCUSSION THREAD

**Discussion by 01fiore** [mocaccinoOS/mocaccino#77] [General] [answered]
the system does not update is by this error:

 WARNING   failed downloading image quay.io/mocaccino/desktop:system-x-layers-0.20220506-3: lchown /var/tmp/luet/image3476682693/usr/lib/libpcre2-8.so.0.10.4: no such file or directory                          
panic: runtime error: invalid memory address or nil pointer dereference                                                                                                                                           
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0xb87ad4]

goroutine 65 [running]:
github.com/mudler/luet/pkg/api/core/types/artifact.(*Checksums).generateSum(0xc007633d20, 0x0?, {{0x103c750?, 0xc009dab980?}, {0xea825b?, 0xc90d05?}})
        /luetbuild/luet/pkg/api/core/types/artifact/checksum.go:78 +0x74
github.com/mudler/luet/pkg/api/core/types/artifact.(*Checksums).generateSHA256(0x0?, 0x0?)
        /luetbuild/luet/pkg/api/core/types/artifact/checksum.go:73 +0x5c
github.com/mudler/luet/pkg/api/core/types/artifact.(*Checksums).Generate(...)
        /luetbuild/luet/pkg/api/core/types/artifact/checksum.go:60
github.com/mudler/luet/pkg/api/core/types/artifact.(*PackageArtifact).Verify(0x0)
        /luetbuild/luet/pkg/api/core/types/artifact/artifact.go:105 +0x8e
github.com/mudler/luet/pkg/installer.(*LuetInstaller).getPackage(0xc00be88160?, {0xc00cf4a8c0?, 0xc006b70000?, {0x1039b90?, 0xc0016a8080?}}, {0x1042be8?, 0xc00bfe6000?})
        /luetbuild/luet/pkg/installer/installer.go:958 +0x67
github.com/mudler/luet/pkg/installer.(*LuetInstaller).downloadWorker(0xc0003d7f00, 0xd59da0?, 0xc0002d7fa0?, 0xc00be88160, 0x0?, {0x1042be8, 0xc00bfe6000})
        /luetbuild/luet/pkg/installer/installer.go:992 +0x12b
created by github.com/mudler/luet/pkg/installer.(*LuetInstaller).download
        /luetbuild/luet/pkg/installer/installer.go:644 +0x4ec

Place the repositories on my system:

# luet repo list
luet
  Luet official Repository
  Revision 1 - 2022-05-03 19:32:54 +0200 CEST
mocaccino-community
  MocaccinoOS community Repository
  Revision 501 - 2022-05-14 09:32:11 +0200 CEST
mocaccino-desktop-stable
  MocaccinoOS desktop Repository (stable)
mocaccino-extra-stable
  MocaccinoOS extra Repository
  Revision 108 - 2022-05-09 21:04:02 +0200 CEST
mocaccino-os-commons-stable
  MocaccinoOS Commons repository
  Revision 31 - 2022-05-13 23:46:02 +0200 CEST
mocaccino-repository-index
  MocaccinoOS Repository Index
  Revision 1 - 2022-05-12 19:47:05 +0200 CEST

Best regards.

**Comment by joostruis** [comment, marked as answer]
That file is part of layers/system-x (core layer)
Not sure how you ended up missing that. Does reinstalling that layer work?

`luet reinstall layers/system-x`

**Reply by 01fiore** [reply]
I installed the package, I turned the computer back on, it no longer sees the eternal port, in the connection menu of Kde.

impossible de charge le module exter à lot de kcm_networkmanager 

library /usr/lib64/qt5/plugin/kcm_networkmanager

I'm outside.

**Reply by 01fiore** [reply]
ethernet sorry.

**Reply by joostruis** [reply]
It is not that there is something missing in the layer/package itself. Something happened on your system that is causing the installation being corrupted. Did you perhaps run out of disk space while upgrading?

What you could do to try  fix this is chroot into the root partition from a live desktop and then try to upgrade again. (or reinstall layers/system-x)

**Reply by 01fiore** [reply]
I installed the SO on a 60 GB partition, I reinstall the macconcino with a new iso next month, I keep you up to date.
Merci pour l'aide à bientôt, tchao.

**Reply by 01fiore** [reply]
In Chroot I can't update the system

mocaccino /home/mocaccino # mount /dev/sde1 /mnt
mocaccino /home/mocaccino # cd /mnt
mocaccino /mnt # mount -t proc proc /mnt/proc
mocaccino /mnt # mount -t sysfs sys /mnt/sys
mocaccino /mnt # mount -o bind /dev /mnt/dev
mocaccino /mnt # mount -t devpts pts /mnt/dev/pts/
mocaccino /mnt # chroot /mnt
mocaccino / # luet upgrade
failed to acquire exclusive app lock: open /var/lock/luet.lock: no such file or directory
mocaccino / # luet search system-profiles
 INFO   Downloaded repository.yaml of 0.00 MB ( 0.00 MiB/s )                                                  
 INFO   Downloading quay.io/mocaccino/mocaccino-community:repository.yaml                                     
 INFO   Pulled: sha256:3130955bae35ede5343996e0da04bc453893b9fbf57f46f7a94dedbb0f060716                       
 INFO   Size: 634B                                                                                            
 INFO   Downloading quay.io/mocaccino/mocaccino-community:tree.tar.gz                                         
 INFO   Pulled: sha256:1f25e0803dadbc2ced5131b7bdb260c583fe7d1117a1449bae915cd5acfb8382                       
 INFO   Size: 81.7KiB                                                                                         
 INFO   Downloading quay.io/mocaccino/mocaccino-community:repository.meta.yaml.tar                            
 INFO   Pulled: sha256:17c46aaf8a02c308e6f5116e5054ba07c7925d882474ae3e71acf8fc2500beaa                       
 INFO   Size: 9.207MiB                                                                                        
 INFO   🏠  Repository mocaccino-community revision: 506 (2022-05-15 20:16:06 +0200 CEST)                     
 INFO   ℹ  Repository: mocaccino-community Priority: 50 Type: docker                                          
 INFO   Downloading quay.io/mocaccino/desktop:20220512170945-repository.yaml                                  
 INFO   Pulled: sha256:c7853c5afff63843f38efbea00955f10b3a5a133fb7d3208e66c230e638078c2                       
 INFO   Size: 679B                                                                                            
 INFO   Downloading quay.io/mocaccino/extra:repository.yaml                                                   
 INFO   Pulled: sha256:8b3b5a8f2f542c2c0d951832c98da30bfba8ba928e3b8027ae00a1b54dcf7add                       
 INFO   Size: 630B                                                                                            
 INFO   Downloading quay.io/mocaccino/os-commons:repository.yaml                                              
 INFO   Pulled: sha256:f7f8b8feb102606b99a62b3addc57fbd0cbfea6b12b14670aea1d42f37e71770                       
 INFO   Size: 632B                                                                                            
 INFO   Downloaded repository.yaml of 0.00 MB ( 0.00 MiB/s )                                                  
 INFO   Downloaded tree.tar.gz of 0.00 MB ( 0.01 MiB/s )                                                      
 INFO   Downloaded repository.meta.yaml.tar.gz of 0.00 MB ( 0.01 MiB/s )                                      
 INFO   🏠  Repository mocaccino-repository-index revision: 1 (2022-05-15 14:42:42 +0200 CEST)                
 INFO   ℹ  Repository: mocaccino-repository-index Priority: 1 Type: http                                      
 INFO   --- Search results (system-profiles): ---    

# luet repo update && luet upgrade -y
 INFO   Downloaded repository.yaml of 0.00 MB ( 0.00 MiB/s )
 INFO   Downloaded tree.tar.zst of 0.00 MB ( 0.00 MiB/s )
 INFO   Downloaded repository.meta.yaml.tar.zst of 0.00 MB ( 0.01 MiB/s )
 INFO   🏠  Repository luet-official revision: 1 (2022-05-03 19:32:54 +0200 CEST)
 INFO   ℹ  Repository: luet Priority: 1 Type: http
 INFO   Downloading quay.io/mocaccino/mocaccino-community:repository.yaml
 INFO   Pulled: sha256:3130955bae35ede5343996e0da04bc453893b9fbf57f46f7a94dedbb0f060716
 INFO   Size: 634B
 INFO   Downloading quay.io/mocaccino/mocaccino-community:tree.tar.gz
 INFO   Pulled: sha256:1f25e0803dadbc2ced5131b7bdb260c583fe7d1117a1449bae915cd5acfb8382
 INFO   Size: 81.7KiB
 INFO   Downloading quay.io/mocaccino/mocaccino-community:repository.meta.yaml.tar
 INFO   Pulled: sha256:17c46aaf8a02c308e6f5116e5054ba07c7925d882474ae3e71acf8fc2500beaa
 INFO   Size: 9.207MiB
 INFO   🏠  Repository mocaccino-community revision: 506 (2022-05-15 20:16:06 +0200 CEST)
 INFO   ℹ  Repository: mocaccino-community Priority: 50 Type: docker
 INFO   Downloading quay.io/mocaccino/desktop:20220512170945-repository.yaml
 INFO   Pulled: sha256:c7853c5afff63843f38efbea00955f10b3a5a133fb7d3208e66c230e638078c2
 INFO   Size: 679B
 INFO   Downloading quay.io/mocaccino/desktop:20220512170945-tree.tar.gz
 INFO   Pulled: sha256:e8bc290fdd32e12c20d77302f1e945d6625cbafd4bb947786b5682142b234000
 INFO   Size: 42.54KiB
 INFO   Downloading quay.io/mocaccino/desktop:20220512170945-repository.meta.yaml.tar
 INFO   Pulled: sha256:a694f6a0dc07b82bcdd760afe5c8b694ca1f2c6bbed79bd739be94d0402896cb
 INFO   Size: 47.56MiB
 INFO   🏠  Repository MocaccinoOS Desktop revision: 356 (2022-05-12 19:09:45 +0200 CEST)
 INFO   ℹ  Repository: mocaccino-desktop-stable Priority: 3 Type: docker
 INFO   Downloading quay.io/mocaccino/extra:repository.yaml
 INFO   Pulled: sha256:8b3b5a8f2f542c2c0d951832c98da30bfba8ba928e3b8027ae00a1b54dcf7add
 INFO   Size: 630B
 INFO   Downloading quay.io/mocaccino/extra:tree.tar.gz
 INFO   Pulled: sha256:ba097f6898efc00dea89a48ca241daeaf88f1da9cbcbbe683118b85a20fefd4b
 INFO   Size: 6.696KiB
 INFO   Downloading quay.io/mocaccino/extra:repository.meta.yaml.tar
 INFO   Pulled: sha256:9931115d5c7a5bc1f82a9d06ff54e1e14bfe60a611f1e4f7c13b4b3cc6ce2bdb
 INFO   Size: 741KiB
 INFO   🏠  Repository mocaccino-extra revision: 108 (2022-05-09 21:04:02 +0200 CEST)
 INFO   ℹ  Repository: mocaccino-extra-stable Priority: 2 Type: docker
 INFO   Downloading quay.io/mocaccino/os-commons:repository.yaml
 INFO   Pulled: sha256:f7f8b8feb102606b99a62b3addc57fbd0cbfea6b12b14670aea1d42f37e71770
 INFO   Size: 632B
 INFO   Downloading quay.io/mocaccino/os-commons:tree.tar.gz
 INFO   Pulled: sha256:b5ef88a24996340f25fab552119080e7a60ec744ceeb70e822c137d9784de95f
 INFO   Size: 26.61KiB
 INFO   Downloading quay.io/mocaccino/os-commons:repository.meta.yaml.tar
 INFO   Pulled: sha256:1d6f8fbcff0fa7d4fb44f3288bea986f8aa889dc185829bfcd01fe37faea8077
 INFO   Size: 973.5KiB
 INFO   🏠  Repository example-repository revision: 31 (2022-05-13 23:46:02 +0200 CEST)
 INFO   ℹ  Repository: mocaccino-os-commons-stable Priority: 2 Type: docker
 INFO   Downloaded repository.yaml of 0.00 MB ( 0.00 MiB/s )
 INFO   Downloaded tree.tar.gz of 0.00 MB ( 0.01 MiB/s )
 INFO   Downloaded repository.meta.yaml.tar.gz of 0.00 MB ( 0.01 MiB/s )
 INFO   🏠  Repository mocaccino-repository-index revision: 1 (2022-05-15 14:42:42 +0200 CEST)
 INFO   ℹ  Repository: mocaccino-repository-index Priority: 1 Type: http
failed to acquire exclusive app lock: open /var/lock/luet.lock: no such file or directory

Why ?

**Reply by joostruis** [reply]
No idea why. But perhaps you should create /var/lock directory because it seems to be missing.
